src/object-store/tests/object_store_test.rs

```diff
@@ -248,7 +248,7 @@ async fn test_file_backend_with_lru_cache() -> Result<()> {
     test_object_crud(&store).await?;
     test_object_list(&store).await?;
 
-    assert_eq!(cache_layer.read_cache_stat().await, (4, 0));
+    assert_eq!(cache_layer.read_cache_stat().await, (0, 0));
```

Comment:
Sorry, but why this stat info is changed?

Reply:
It won't go into the cache if the read is without deterministic range size(i.e., `read_with(p2).range(0..4)`).

```rust
    // Read the deleted file without a deterministic range size requires an extra `stat.`
    // Therefore, it won't go into the cache.
    assert!(store.read(p2).await.is_err());

    let p3 = "test_file3";
	@@ -376,13 +379,20 @@ async fn test_object_store_cache_policy() -> Result<()> {
    let _ = store.read(p3).await.unwrap();
    let _ = store.read_with(p3).range(0..5).await.unwrap();

    assert_eq!(cache_layer.read_cache_stat().await, (3, 35));

    // However, The real open file happens after the reader is created.
    // The reader will throw an error during the reading
    // instead of returning `NotFound` during the reader creation.
    // The entry count is 4, because we have the p2 `NotFound` cache.
    assert!(store.read_with(p2).range(0..4).await.is_err());
    assert_eq!(cache_layer.read_cache_stat().await, (4, 35));
```